app/connectors/cache/MongoCacheConnector.scala

```diff
@@ -56,7 +58,14 @@ class MongoCacheConnector @Inject()(cacheClientFactory: MongoCacheClientFactory,
   def save[T](key: String, data: T)(implicit authContext: AuthContext, hc: HeaderCarrier, format: Format[T]): Future[CacheMap] = {
     authConnector.getCredId flatMap {
       credId => {
-        mongoCache.createOrUpdate(credId, authContext.user.oid, data, key).map(toCacheMap)
+        mongoCache.createOrUpdate(credId, Some(authContext.user.oid), data, key).map(toCacheMap)
+      }
+    }
+  }
+  def save[T](id: String, key: String, data: T)(implicit hc: HeaderCarrier, format: Format[T]): Future[CacheMap] = {
+    authConnector.getCredId flatMap {
+      credId => {
+        mongoCache.createOrUpdate(credId, None, data, key).map(toCacheMap)
```

Comment:
here we are asking authConnector for credId, but id is credId?

Reply:
Will take a 👀